tests/dma/vif/stcycl.cpp

```diff
@@ -106,53 +105,63 @@ void testCycleTruncated(u32 cl, u32 wl) {
 			vifcode.Data32(i);
 		}
 
-		DMA::SendSimple(DMA::D0, vifcode.Raw(), 16 * vifPacketSize);
+		DMA::SendSimple(context->dmaChannel, vifcode.Raw(), 16 * vifPacketSize);
 	}
 	
-	SyncDCache(vu0_mem, vu0_mem + 16 * displaySize);
+	SyncDCache(vuMem, vuMem + 16 * displaySize);
 	
-	u32 cycleReg = *R_EE_VIF0_CYCLE;
+	u32 cycleReg = context->vifRegs->cycle;
 	printf("CYCLE: %08x, ", cycleReg);
 	
 	for (u32 i = 0; i < displaySize / 4; i++) {
 		printf("%x%x%x%x",  
-			vu0_32[(i * 0x10) + 0x0], vu0_32[(i * 0x10) + 0x4], vu0_32[(i * 0x10) + 0x8], vu0_32[(i * 0x10) + 0xC]);
+			vuMem_32[(i * 0x10) + 0x0], vuMem_32[(i * 0x10) + 0x4], vuMem_32[(i * 0x10) + 0x8], vuMem_32[(i * 0x10) + 0xC]);
 	}
 	
 	printf("\n");
 }
 
-int main(int argc, char *argv[]) {
-	printf("-- TEST BEGIN\n");
-	
+void doTest(TEST_CONTEXT* context) {
 	for(u32 cl = 0; cl <= 8; cl++) {
 		for(u32 wl = 0; wl <= 8; wl++) {
 			printf("normal (cl %d, wl %d): ", cl, wl);
-			testCycleNormal(cl, wl);
+			testCycleNormal(context, cl, wl);
 		}
 	}
 	
 	for(u32 cl = 251; cl <= 255; cl++) {
 		for(u32 wl = 0; wl <= 8; wl++) {
 			printf("normal (cl %d, wl %d): ", cl, wl);
-			testCycleNormal(cl, wl);
+			testCycleNormal(context, cl, wl);
 		}
 	}
 	
 	for(u32 cl = 0; cl <= 8; cl++) {
 		for(u32 wl = 251; wl <= 255; wl++) {
 			printf("normal (cl %d, wl %d): ", cl, wl);
-			testCycleNormal(cl, wl);
+			testCycleNormal(context, cl, wl);
 		}
 	}
 	
 	for(u32 cl = 0; cl <= 8; cl++) {
 		for(u32 wl = 0; wl <= 8; wl++) {
 			printf("truncated (cl %d, wl %d): ", cl, wl);
-			testCycleTruncated(cl, wl);
+			testCycleTruncated(context, cl, wl);
 		}
 	}
+}
+
+int main(int argc, char *argv[]) {
+	printf("-- TEST BEGIN\n");
+	
+	printf("VIF0 -------------------------------------------------------------\n");
```

Comment:
66 total characters / 61 hyphens seems fairly arbitrary.  Not that it really matters.  I'd personally prefer maybe a more markdown like style like "== VIF0 ==" or something, or just a smaller number of hyphens.  Otherwise it'll probably end up inconsistent...

Again, not like it really matters.

-[Unknown]